migrations/versions/2ee66194b9eb_remove_run_table.py

```diff
@@ -17,24 +17,36 @@
 
 
 def upgrade():
-    # TODO: data migration from run to benchmark_result goes here!
-
     op.add_column(
         "benchmark_result",
-        sa.Column("run_tags", postgresql.JSONB(astext_type=sa.Text()), nullable=False),
+        sa.Column("run_tags", postgresql.JSONB(astext_type=sa.Text()), nullable=True),
     )
     op.add_column("benchmark_result", sa.Column("run_reason", sa.Text(), nullable=True))
     op.add_column(
         "benchmark_result", sa.Column("commit_id", sa.String(length=50), nullable=True)
     )
     op.add_column(
         "benchmark_result",
-        sa.Column("hardware_id", sa.String(length=50), nullable=False),
+        sa.Column("hardware_id", sa.String(length=50), nullable=True),
     )
     op.drop_constraint("summary_run_id_fkey", "benchmark_result", type_="foreignkey")
     op.create_foreign_key(None, "benchmark_result", "hardware", ["hardware_id"], ["id"])
     op.create_foreign_key(None, "benchmark_result", "commit", ["commit_id"], ["id"])
 
+    op.execute(
+        """
+        UPDATE benchmark_result SET
+            run_tags = jsonb_build_object('name', coalesce(run.name, '')),
+            run_reason = run.reason,
+            commit_id = run.commit_id,
+            hardware_id = run.hardware_id
+        FROM run
+        WHERE benchmark_result.run_id = run.id
+        """
+    )
+    op.alter_column("benchmark_result", "run_tags", nullable=False)
```

Comment:
So if `run_name` isn't specified this will just be `{}`? I thought JP was against that for perf reasons, though we're already doing that elsewhere, so if we want to change maybe we should do a larger cleanup

Reply:
See my response [here](https://github.com/conbench/conbench/pull/1437#discussion_r1296011456) -- basically I think storing `null`s here would make our lives harder rather than easier.